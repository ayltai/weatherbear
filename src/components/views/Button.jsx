import { IconButton, makeStyles, Tooltip, } from '@material-ui/core';
import PropTypes from 'prop-types';
import React from 'react';
import { useTranslation, } from 'react-i18next';

export const Button = props => {
    const classes = makeStyles(theme => ({
        container : {
            padding : theme.spacing(1),
        },
    }))();

    const { t, } = useTranslation();

    return (
        <Tooltip title={t(props.tooltip)}>
            <IconButton
                className={classes.container}
                edg={props.edge}
                size={props.size}
                onClick={() => {
                    if (props.onClick) props.onClick();
                }}>
                {props.icon}
            </IconButton>
        </Tooltip>
    );
};

Button.propTypes = {
    className : PropTypes.string,
    tooltip   : PropTypes.string,
    icon      : PropTypes.node,
    edge      : PropTypes.oneOf([
        'end',
        'start',
    ]),
    size      : PropTypes.oneOf([
        'small',
        'medium',
    ]),
    onClick   : PropTypes.func,
};
