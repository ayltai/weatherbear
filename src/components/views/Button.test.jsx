import { IconButton, } from '@material-ui/core';
import { mount, shallow, } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import { Button, } from './Button';

describe('<Button />', () => {
    it('renders correctly', () => {
        expect(renderer.create(<Button />).toJSON()).toMatchSnapshot();
    });

    it('mounts without error', () => {
        mount(<Button />);
    });

    it('triggers onClick when clicked', () => {
        const onClick = jest.fn();

        shallow(<Button onClick={onClick} />)
            .find(IconButton)
            .simulate('click');

        expect(onClick.mock.calls.length).toEqual(1);
    });
});
