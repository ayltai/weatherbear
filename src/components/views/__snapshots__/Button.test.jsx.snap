// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<button
  aria-describedby={null}
  className="MuiButtonBase-root MuiIconButton-root makeStyles-container-1"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  title={null}
  type="button"
>
  <span
    className="MuiIconButton-label"
  />
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;
